src/reader.rs

```diff
@@ -4796,35 +4796,64 @@ fn get_autosuggestion_performer(
                 tokens.first().map(|tok| tok.offset()).unwrap_or(cursor_pos),
             ) == search_string_range
         };
+
+        // Only to be used if no case-sensitive suggestions are found.
+        let mut icase_history_result = None;
+
         if cursor_line_has_process_start {
-            let mut searcher = HistorySearch::new_with_type(
+            let mut searcher = HistorySearch::new_with(
                 history,
                 search_string.to_owned(),
                 SearchType::LinePrefix,
+                SearchFlags::IGNORE_CASE,
+                0,
             );
+
             while !ctx.check_cancel() && searcher.go_to_next_match(SearchDirection::Backward) {
                 let item = searcher.current_item();
 
-                // Suggest only a single line each time.
-                let matched_line = item
+                // The history item's may have multiple lines of text.
+                // Only suggest the line that actually contains the search string.
+
+                let lines = item
```

Comment:
FWIW I got some upcoming changes that will allow multiline commands to be autosuggested 
(and if there is no full match, it will fall back to the existing line-by-line logic),
but that should be independent of your change.